### DISCUSSION TITLE
Access modifier to restrict use only in the current partial declaration

### DISCUSSION THREAD

**Discussion by alrz** [dotnet/csharplang#4529] [Language Ideas] [answered]
This would be much similar to `fileprivate` access modifier in Swift, but instead of the "file", they would be only accessible to the current `partial` declaration.

```cs
partial class C {
  int Method() { .. }
}

// Generated
partial class C {
  hidden int Method() { .. } // does not conflict with `Method` in user-code
} 
```
Beyond code generation, it's also useful to have a more restrictive member access modifier for types that span multiple files.

Sometimes it doesn't help to make a new class for members that you want to separate out from the rest of the declaration. Using `private` members still doesn't prohibit use in other parts and therefore may break the separation you intended to enforce.

**Comment by Joe4evr** [comment, marked as answer]
Duplicate of #1288
